modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Migrations.scala

```diff
@@ -0,0 +1,223 @@
+package com.snowplowanalytics.iglu.schemaddl.parquet
+
+import com.snowplowanalytics.iglu.schemaddl.parquet.Type.{Array, Struct, DecimalPrecision}
+
+/*
+  Parquet schemas are migrated though the `merge.schema = true` loading option in Spark/Databricks.
+  Migrations in this package do not define spark 3.0 style `ATLER TABLE` transformations, similar to Redshift. Instead
+  they are used to categorize schema changes into 3 categories:
+   - Patch - Miscellaneous changes that would be handled by the `merge.schema`:
+      * `NullableRequired` - nullable to required change
+      * `NestedKeyAddition` - addition of the new Struct key anywhere except for top level Schema.
+      * `TypeWidening` - Compatible type casting. Such as Int -> Long (cast as BitInt) -> Decimal -> Double.
+   - Minor - Reserved for key addition in top level schema `TopLevelKeyAddition`.
+   - Major - Breaking change that would either lead to oen of these outcomes:
+      * `KeyRemoval` historical data loss, such as removing field in Struct. Loading will succeed with data loss.
+      * `IncompatibleType` Incompatible type casting, such as String -> Int, etc. Loading will fail.
+      * `RequiredNullable` Nullable to required field or array migration. Loading will fail.
+
+   Note that JSON type and String are interchangeable, in the context of this library because Spark does not support
+   Parquet with JSON type, so they are stringified.
```

Comment:
True, Json and string are interchangeable from pov of databricks/spark.  I will just point out a slight different from point of view of the consumer:

The transformer treats the string/json parquet types slightly differently.  Imagine the incoming JSON contains a key/value like `{"foo": "bar"}`:

- If field `foo` has type=string then it writes `bar` to the parquet file
- If field `foo` has type=json then it writes `"bar"` to the parquet file (including quote marks).  This is because `"bar"` is valid json.

From consumer's point of view, there is a guarantee that you can json-deserialize the quoted string to get the original value.

But this is a minor detail -- we have bigger problems to worry about before thinking about this detail!!